### ISSUE TITLE
Noise Plugin -- java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread

### ISSUE THREAD

**Issue by bardram** [cph-cachet/flutter-plugins#22]
The Noise Plugin crashes with a `java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread` exception:

```
E/AndroidRuntime( 4300): Process: dk.cachet.carpmobilesensingapp, PID: 4300
E/AndroidRuntime( 4300): java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: Thread-6
E/AndroidRuntime( 4300): 	at io.flutter.embedding.engine.FlutterJNI.ensureRunningOnMainThread(FlutterJNI.java:794)
E/AndroidRuntime( 4300): 	at io.flutter.embedding.engine.FlutterJNI.dispatchPlatformMessage(FlutterJNI.java:684)
E/AndroidRuntime( 4300): 	at io.flutter.embedding.engine.dart.DartMessenger.send(DartMessenger.java:80)
E/AndroidRuntime( 4300): 	at io.flutter.embedding.engine.dart.DartExecutor.send(DartExecutor.java:174)
E/AndroidRuntime( 4300): 	at io.flutter.view.FlutterNativeView.send(FlutterNativeView.java:144)
E/AndroidRuntime( 4300): 	at io.flutter.plugin.common.EventChannel$IncomingStreamRequestHandler$EventSinkImplementation.success(EventChannel.java:226)
E/AndroidRuntime( 4300): 	at cachet.sensors.noisemeter.NoiseMeterPlugin$1.run(NoiseMeterPlugin.java:137)
E/AndroidRuntime( 4300): 	at java.lang.Thread.run(Thread.java:764)
```

**Comment by bardram** [comment]
A similar exception has been reported in many other Flutter plugins lately. See e.g.

* https://github.com/dooboolab/flutter_sound/issues/93
* https://github.com/flutter/flutter/issues/32315
* https://github.com/g123k/flutter_plugin_device_apps/issues/11

**Comment by bardram** [comment]
I fixed this problem in the [esense](https://github.com/cph-cachet/flutter-plugins/tree/master/packages/esense) plugin. You need to wrap methods/event channels. 

See

* https://github.com/cph-cachet/flutter-plugins/blob/master/packages/esense/android/src/main/java/dk/cachet/esense/MainThreadEventSink.java
* https://github.com/cph-cachet/flutter-plugins/blob/master/packages/esense/android/src/main/java/dk/cachet/esense/MainThreadResult.java

**Comment by PatriceG** [comment]
Hi,
I fixed it with the following changes in NoiseMeterPlugin.listen()
It's working in the emulator.
I don't have much time to submit a merge request for now but here is the modified method for a quick fix.
The trick is to use a Handler to have the eventSink method run on the UI thread.
The noiselevel variable also has to be set final:

```
/**
     * Listens to the [MediaRecorder] object with a given frequency and creates a callback
     * to the [NoiseLevel] object in Flutter which created this Plugin object.
     */
    private void listen() {
        Log.d(TAG, "listen()");

        Thread thread = new Thread(new Runnable() {
            @Override
            public void run() {
                while (isRecording && eventSink != null) {
                    try {
                        int volume = mediaRecorder.getMaxAmplitude();  //Get the sound pressure value
                        final NoiseLevel noiseLevel = new NoiseLevel(volume);
                        new Handler(Looper.getMainLooper()).post(new Runnable() {
                          @Override
                          public void run() {
                            eventSink.success(noiseLevel.getDecibel());
                          }
                       });   
                        Thread.sleep(frequency);
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }
            }
        });
        thread.start();
    }
```

**Comment by bardram** [comment]
Thanks @PatriceG -- I've implemented your bug fix and released a version 1.1. to pub.dev.
